### ISSUE TITLE
Re-fix Skia color correctness

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#10710]
We are about to [roll back](https://github.com/flutter/engine/pull/3775) the Skia color-correctness fix. It causes frequent reboots on (at least) iPhone 6 model A1586, iPhone 6 model A1549, and iPod Touch model A1574.

[More info](https://docs.google.com/document/d/12WpzYF0nUT8VfA7JDx9OwZa4H3sd9DKacm04-dBAppw/edit)

@chinmaygarde @goderbauer @jason-simmons @mattsarett

**Comment by Hixie** [comment]
Original PR was: https://github.com/flutter/engine/pull/3743

**Comment by mattsarett** [comment]
Sounds good to me.  Do you have any logs or data that might help us diagnose the cause of the issues?

**Comment by mattsarett** [comment]
@brianosman

**Comment by yjbanov** [comment]
Is the "More info" link in the description helpful at all? If not, we have plenty of devices in the office that can reproduce this.

**Comment by mattsarett** [comment]
Ahh thanks, sorry I missed that.

**Comment by lukef** [comment]
Not sure if you want to piggy back on this issue but this issue is causing incorrect rendering of widgets in the Android emulators on macOS. I can open a new issue if you prefer?

**Comment by mattsarett** [comment]
This issue works for me.  As far as I know, we are now tracking three things:
(1) Black screen on Android emulator on Mac
(2) Frequent reboots on iPhone and iPod
(3) Colors possibly wrong on Android widget

**Comment by lukef** [comment]
Not sure if it is related but I have had some `SurfaceView` crashes on Android also. See: https://github.com/flutter/flutter/issues/10709.

**Comment by mattsarett** [comment]
I'm not sure that I have enough information to determine if the SurfaceView crashes are related.  I would be interested to see what you mean regarding the wrong colors on Android widgets.

**Comment by mattsarett** [comment]
I've determined the cause of (1).  I'm going to upload another PR with that fix.

Still working on (2) and potentially (3) - @lukef can you provide more detail on what you are seeing?

**Comment by mattsarett** [comment]
`flutter drive --profile -t test_driver/transitions_perf.dart -d X`

The above command seems to be failing when run with a local engine.  Any ideas?  Does this work for anyone else?

We're feeling a little blocked by this - I think we're definitely going to need to use a local engine to dig into what is going wrong with (2).

**Comment by eseidelGoogle** [comment]
The bugs associated with @mattsarett's list:
1. black screens: https://github.com/flutter/flutter/issues/10617
2. device-lab reboots: I'm not sure if this ever got a bug @yjbanov?
3. widget colors wrong -- this doesn't have a bug yet, @Hixie and I are tracking via email.  I suspect it's an incarnation of https://github.com/flutter/flutter/issues/10655 except that @lukef's widgets were written while your patch were in, so they now look wrong with it out.  But it also may be unrelated as @lukef's issue only occurs on emulators.

**Comment by goderbauer** [comment]
@mattsarett Running with a local engine on iOS worked fine for me yesterday. I was using the following command:
```
flutter drive --profile -t test_driver/transitions_perf.dart -d i --local-engine=ios_profile --trace-startup
```

**Comment by goderbauer** [comment]
@eseidel 
> 2. device-lab reboots: I'm not sure if this ever got a bug @yjbanov?

We never filed a bug for that, but we captured [crash logs](https://docs.google.com/document/d/12WpzYF0nUT8VfA7JDx9OwZa4H3sd9DKacm04-dBAppw/edit).

**Comment by yjbanov** [comment]
@eseidel @goderbauer 

> device-lab reboots: I'm not sure if this ever got a bug @yjbanov?

It's this bug :)

**Comment by brianosman** [comment]
Okay, I was able to flutter drive with local-engine by removing --profile. I think the mis-match between --profile and my build type (ios_debug_unopt) is perhaps to blame?

**Comment by mattsarett** [comment]
Great thanks, that fixes me as well.  FWIW, I've not been able to reproduce in about 6 runs on the iPad Pro, which makes me think this might be specific to older devices.

**Comment by goderbauer** [comment]
@mattsarett That is my suspicion as well. We saw it on the (older) devices Yegor mentioned in the first post, but it didn't reproduce on an iPhone 6S.

**Comment by Hixie** [comment]
@brianosman The build you specify has to match the --profile/--release/--debug mode you pick and the target host's architecture.

**Comment by chinmaygarde** [comment]
This was fixed in and landed https://github.com/flutter/engine/pull/3826
